plugins/modules/ndo_l3out_template.py

```diff
@@ -0,0 +1,900 @@
+
+DOCUMENTATION = r"""
+---
+module: ndo_l3out_template
+short_description: Manage L3Outs on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage L3Outs on Cisco Nexus Dashboard Orchestrator (NDO).
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  template:
+    description:
+    - The name of the L3Out template.
+    type: str
+    aliases: [ l3out_template ]
+    required: true
+  name:
+    description:
+    - The name of the L3Out.
+    type: str
+  uuid:
+    description:
+    - The UUID of the L3Out.
+    - This parameter is required when the L3Out needs to be updated.
+    type: str
+  description:
+    description:
+    - The description of the L3Out.
+    type: str
+  vrf:
+    description:
+    - The VRF associated with the L3Out.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the VRF.
+        required: true
+        type: str
+      schema:
+        description:
+        - The name of the schema.
+        required: true
+        type: str
+      template:
+        description:
+        - The name of the template.
+        required: true
+        type: str
+  l3_domain:
+    description:
+    - The name of the L3 Domain.
+    type: str
+  target_dscp:
+    description:
+    - The DSCP Level of the L3Out.
+    type: str
+    choices:
+      - af11
+      - af12
+      - af13
+      - af21
+      - af22
+      - af23
+      - af31
+      - af32
+      - af33
+      - af41
+      - af42
+      - af43
+      - cs0
+      - cs1
+      - cs2
+      - cs3
+      - cs4
+      - cs5
+      - cs6
+      - cs7
+      - expedited_forwarding
+      - unspecified
+      - voice_admit
+  pim:
+    description:
+    - The protocol independent multicast (PIM) flag of the L3Out.
+    - By default, PIM is disabled. To enable the PIM, Layer 3 Multicast must be enabled on the O(vrf).
+    type: bool
+  interleak:
+    description:
+    - The name of the Route Map Policy for Route Control that needs to be associated with Interleak route map.
+    type: str
+  static_route_redistribution:
+    description:
+    - The name of the Route Map Policy for Route Control that needs to be associated with Static Route Redistribution route map.
+    type: str
+    aliases: [ static_route ]
+  connected_route_redistribution:
+    description:
+    - The name of the Route Map Policy for Route Control that needs to be associated with Connected Route Redistribution route map.
+    type: str
+    aliases: [ connected_route ]
+  attached_host_route_redistribution:
+    description:
+    - The name of the Route Map Policy for Route Control that needs to be associated with Attached Host Route Redistribution route map.
+    type: str
+    aliases: [ attached_host_route ]
+  bgp:
+    description:
+    - The BGP routing protocol configuration of the L3Out.
+    - Use empty dict ({}) to clear the complete BGP configuration.
+    - Use empty string ("") to clear the BGP sub options configuration.
```

Comment:
Should this be mentioned at the sub-option configuration level?
```suggestion
    - Use empty string ("") to clear the BGP sub options configuration.
```